lib/compeon/api_client/middlewares/jsonapi_response.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require_relative 'deep_hash_transformer'
+
+require 'json'
+
+module Compeon
+  class APIClient
+    module Middlewares
+      class JSONAPIResponseMiddleware < Faraday::Response::Middleware
+        include DeepHashTransformer
+
+        Faraday::Response.register_middleware jsonapi_response: self
```

Comment:
Will this result in Faraday registering this as a default middleware for every client? If yes, that could interfer with other Faraday clients.